src/foundry/client/pixi/layers/placeables/drawings.d.ts

```diff
@@ -90,34 +91,58 @@ declare global {
     }
   }
 
-  type NewDrawingData = ClientSettings.Values["core.defaultDrawingConfig"] &
-    (
-      | {
-          type: typeof foundry.CONST.DRAWING_TYPES.RECTANGLE | typeof foundry.CONST.DRAWING_TYPES.ELLIPSE;
-          points: [];
-        }
+  type NewDrawingData = ClientSettings.Values["core.defaultDrawingConfig"] & {
+    _id: null;
+    author: string;
+    fillColor: string;
+    strokeColor: string;
+    fontFamily: typeof CONFIG.defaultFontFamily;
+    x: number;
+    y: number | undefined;
+    /** Following properties come from DrawingDocument.CleanData */
+    shape: {
+      height: null;
+      points: PointArray[];
+      radius: null;
+      width: null;
+    };
+    bezierFactor: number;
+    fillAlpha: number;
+  } & (
       | {
-          type: typeof foundry.CONST.DRAWING_TYPES.POLYGON;
-          points: PointArray[];
+          /** case "rect" && case "ellipse" */
+          shape: {
+            type: (typeof Drawing)["SHAPE_TYPES"]["RECTANGLE" | "ELLIPSE"];
+            width: number;
+            height: number;
+          };
         }
       | {
-          type: typeof foundry.CONST.DRAWING_TYPES.FREEHAND;
-          points: PointArray[];
+          /** case "polygon" && case "freehand" */
+          shape: {
+            type: (typeof Drawing)["SHAPE_TYPES"]["POLYGON"];
+            points: PointArray[];
+          };
           bezierFactor: number;
         }
       | {
-          type: typeof foundry.CONST.DRAWING_TYPES.TEXT;
+          /** case "text" */
+          shape: {
+            type: (typeof Drawing)["SHAPE_TYPES"]["RECTANGLE" | "ELLIPSE"];
+            width: number;
+            height: number;
+          };
+          /** @defaultValue `"#FFFFFF"` */
           fillColor: string;
+
+          /** @defaultValue `0.10` */
           fillAlpha: number;
+
+          /** @defaultValue `"#FFFFFF"` */
           strokeColor: string;
+
+          /** @defaultValue `"New Text"` */
           text: string;
         }
-    ) & {
-      author: string;
-      fillColor: string;
-      strokeColor: string;
-      fontFamily: typeof CONFIG.defaultFontFamily;
-      x: number | undefined;
-      y: number | undefined;
-    };
+    ) & {};
```

Comment:
What's up with this `& {}`?

Reply:
Was moving things around and forgot to remove that